crates/bevy_ecs/src/reflect.rs

```diff
@@ -123,6 +142,108 @@ impl<C: Component + Reflect + FromWorld> FromType<C> for ReflectComponent {
     }
 }
 
+/// A struct used to operate on reflected [`Resource`] of a type.
+///
+/// A [`ReflectResource`] for type `T` can be obtained via
+/// [`bevy_reflect::TypeRegistration::data`].
+#[derive(Clone)]
+pub struct ReflectResource {
+    insert_resource: fn(&mut World, &dyn Reflect),
+    apply_resource: fn(&mut World, &dyn Reflect),
+    remove_resource: fn(&mut World),
+    reflect_resource: fn(&World) -> Option<&dyn Reflect>,
+    reflect_resource_unchecked_mut: unsafe fn(&World) -> Option<ReflectMut>,
+    copy_resource: fn(&World, &mut World),
+}
+
+impl ReflectResource {
+    /// Insert a reflected [`Resource`] into the world like [`insert_resource()`](World::insert_resource).
+    pub fn insert_resource(&self, world: &mut World, resource: &dyn Reflect) {
+        (self.insert_resource)(world, resource);
+    }
+
+    /// Uses reflection to set the value of this [`Resource`] type in the world to the given value.
+    ///
+    /// # Panics
+    ///
+    /// Panics if there is no [`Resource`] of the given type.
+    pub fn apply_resource(&self, world: &mut World, resource: &dyn Reflect) {
+        (self.apply_resource)(world, resource);
+    }
+
+    /// Removes this [`Resource`] type from the world.
```

Comment:
Could you add what happens in the case there's no resource to remove?

Reply:
Hm... Nothing happens, but I think we should return the removed resource as Reflect... Let me fix it.

Reply:
Hm... It looks like I can't transform resource into `Reflect`, I only can take a reference. Will only update the description then.

Reply:
Done!